src/common/file_linux.c

```diff
@@ -134,7 +134,8 @@ util_file_dir_next(struct dir_handle *handle, struct file_info *info)
 	struct dirent *d = readdir(handle->dirp);
 	if (d == NULL)
 		return 0; /* break */
-	strcpy(info->filename, d->d_name);
+	strncpy(info->filename, d->d_name, NAME_MAX);
```

Comment:
shouldn't we error out if length of d_name is larger than the buffer? The contents of filename variable is useless anyway in that case (because we lost information).